### PR TITLE
Update compatibility and dependencies for 0.10.0

### LINKED ISSUE #1214: Update compatibility and dependencies for 0.10.0
### Description of the modifications

Remove ANSYS 2022R1 from supported versions starting with `0.10.0`: specify `0.9.0` as maximum `ansys-dpf-core` version to use for `221`.

Update the compatibility page and table to explain that `ansys-dpf-gate`, `ansys-dpf-gatebin` and `ansys-grpc-dpf` are now part of the OS-specific versions of `ansys-dpf-core`.

### Useful links and references

_No response_

### PR DESCRIPTION
**Pull request by PProfizi**
TLDR; `ansys-dpf-gate`, `ansys-dpf-gatebin`, and `ansys-grpc-dpf` no longer exist as separate packages. They are now directly shipped within `ansys-dpf-core` as modules. This PR changes the doc (mainly the compatibility table) accordingly.

Closes #1214

### CODE REVIEW

File: docs/source/getting_started/compatibility.rst [hunk 1]
```diff
@@ -15,84 +15,115 @@ Client-server
...
+``ansys-dpf-gatebin`` and ``ansys-grpc-dpf`` are no longer dependencies and are directly integrated
+within ``ansys-dpf-core`` as modules. This introduced a breaking change to simplify installation
+and prevent synchronization issues between the PyDPF libraries, requiring to drop support for Ansys
+previous to 2022 R2.
+
 **Ansys strongly encourages you to use the latest packages available**, as far they are compatible
-with the Server version you want to use. Using Ansys 2022 R2, if ansys-dpf-core module with
-0.8.0 version is the latest available package, it should be used.
+with the Server version you want to use. Using Ansys 2023 R1 for example, if ansys-dpf-core
+module with 0.10.0 version is the latest available package, it should be used.
```
Comment (rafacanton) [inline docs/source/getting_started/compatibility.rst:37]: This sentence has the verb to use twice. Can you rephrase it?

File: README.md [hunk 2]
```diff
@@ -93,11 +93,25 @@ To install PyDPF-Core with its optional plotting functionalities, use this comma
 
 For more information on PyDPF-Core plotting capabilities, see [Plot](https://dpf.docs.pyansys.com/version/stable/user_guide/plotting.html) in the PyDPF-Core documentation.
 
-To use PyDPF-Core with Ansys 2021 R1, install the latest version
+To use PyDPF-Core with Ansys 2022R1, install the latest compatible version
```
Comment (JennaPaikowsky) [inline README.md:96]:
```suggestion
To use PyDPF-Core with Ansys 2022 R1, install the latest compatible version
```

File: README.md [hunk 3]
```diff
@@ -93,11 +93,25 @@ To install PyDPF-Core with its optional plotting functionalities, use this comma
...
 with this command:
 
 ```con
-   pip install ansys-dpf-core<0.3.0
+   pip install ansys-dpf-core<0.10.0
+```
+
+To use PyDPF-Core with Ansys 2021R2, install the latest compatible version
```
Comment (JennaPaikowsky) [inline README.md:103]:
```suggestion
To use PyDPF-Core with Ansys 2021 R2, install the latest compatible version
```

File: docs/source/getting_started/compatibility.rst (hunk 1 above)
Comment (PProfizi) [inline docs/source/getting_started/compatibility.rst:37]:
```suggestion
with the Server version you want to run. Considering Ansys 2023 R1 for example, if ansys-dpf-core
module with 0.10.0 version is the latest available compatible package, it should be used.
```

**Review by rafacanton — APPROVED** [review]
